modules/internal/ChapelArray.chpl

```diff
@@ -484,13 +488,13 @@ module ChapelArray {
       type elemValType = _getLiteralType(elemVal.type);
 
       if elemKeyType != keyType {
-        compilerError("Associative array key element " + ((i+3)/2):string +
+        compilerError("Associative array key element " + ((i+1)/2):string +
                        " expected to be of type " + keyType:string +
                        " but is of type " + elemKeyType:string);
       }
 
       if elemValType != valType {
-        compilerError("Associative array value element " + ((i+2)/2):string
+        compilerError("Associative array value element " + ((i+1)/2):string
```

Comment:
If I'm understanding this right we want just `(i/2):string` here. I think we want i=1 to to print 0, instead of 1, right?

Reply:
I have to admit, I've mostly treated these as a black box once testing worked, and don't feel confident in them, particularly after the multiple zero-based PRs have messed with them in multiple ways.  Let's see...

i is counting by 2 and aligned to 0, so will only be 0, 2, 4, 6, 8...  But you're right that that suggests that the +1 has become meaningless and that i/2 would give the same result more simply.

(it didn't help that on master in 1.20 the expressions were `(i+2)/2` and `(i+1)/2` even though i was similarly counting by 2's, albeit aligned with 1, 3, 5, ...)

Reply:
Making that change caused the relevant tests to pass as you'd expected.  https://github.com/chapel-lang/chapel/pull/14521/commits/0f341be1c31ed8522e3a4d6f8701cd50b34fe9a3 implements it.